include/common/core/arch_config.hpp

```diff
@@ -83,74 +87,114 @@ struct load_store_attr_t<msg_type::block_2d, gpu_arch::XeLpg>
           msg_type::block_2d,
           gpu_arch::XeLpg> {};
 
+template <gpu_arch arch_tag>
+inline constexpr bool arch_has_2d_load_store() {
+  using block2d_attr = load_store_attr_t<msg_type::block_2d, arch_tag>;
+  return block2d_attr::has_hw_block_2d;
+}
+
```

Comment:
Is it better to
```suggestion
template <gpu_arch arch_tag>
inline constexpr bool arch_has_2d_load_store
  = load_store_attr_t<msg_type::block_2d, arch_tag>::has_hw_block_2d;
```

Similar for `arch_has_xmx`

Reply:
```
template <gpu_arch arch_tag>
	inline constexpr bool arch_has_2d_load_store
```
This is a variable. Is it okay to add template before it?

Reply:
Sure! Since C++14.